doc/advopt.txt

```diff
@@ -29,6 +30,8 @@ Runtime checks (see -x):
 Advanced options:
   -o:FILE, --out:FILE       set the output filename
   --outdir:DIR              set the path where the output file will be written
+  --usenimcache             will use `outdir=$$nimcache`, whichever it resolves
```

Comment:
I don't understand the point of ``--usenimcache``.

Reply:
`--usenimcache` is what's implied by `nim r`, and avoids clobbering your sources with generated files (exe, js, html), or having to deal with gitignore.

It's useful for other commands apart from `nim r`, as it allows specifying to output to nimcache without having to guess and repeat what nimcache will resolve to, eg:
```
nim doc -r --usenimcache main # outputs to $nimcache/main.html
nim doc --usenimcache main # also useful without -r; the hintSuccessX tells you where it lands
nim js --usenimcache main # ditto
```

so if I have `~/.config/nim/config.nims` that contains `--nimcache:/tmp/nimcache`, it'll output to /tmp/nimcache/main.js; without it, i'd need to pass `--outdir:/tmp/nimcache` to achieve same effect.
Since nimcache can be determined by several config files, guessing what nimcache resolves to ahead of time is not desirable.

## future: `--usenimcache:usehash`
In future work, I want to allow the same as what `rdmd` does: store the generated outputs under a dir named after a hash of all options (quite simple to implement on top of what's already implemented in `changeDetectedViaJsonBuildInstructions`, eg `$nimcache/h-$hash/{foo.json,foo}`

## example
```
rdmd -O t10721.d # recompiles (1st time with these options)
rdmd -g t10721.d # recompiles (1st time with these options)
rdmd -O t10721.d # fast: no recompilation (cache hit)
rdmd -g t10721.d # fast: no recompilation (cache hit)
# modify t10721.d or any (recursive) dependency
rdmd -O t10721.d # recompiles (cache miss since hash changed)
```

here's for reference what rdmd outputs after running this:

```
/var/folders/ll/dtd6ksrn50n3xkdy4y61tf_80000gn/T/.rdmd-501/
├── eval.4176BE8C7144BB6EAF3CA1B706BFC853.d
├── eval.D25DF9CF331213004D56C55409E5E99C.d
├── rdmd-eval.4176BE8C7144BB6EAF3CA1B706BFC853.d-1B5EF85CE49FEDB422F66F19721FE135
│   ├── eval.4176BE8C7144BB6EAF3CA1B706BFC853
│   ├── objs
│   ├── rdmd.deps
│   └── rdmd.lock
├── rdmd-eval.D25DF9CF331213004D56C55409E5E99C.d-0BF83D66C1B856F99825200CBECE2C8B
│   ├── eval.D25DF9CF331213004D56C55409E5E99C
│   ├── rdmd.deps
│   └── rdmd.lock
├── rdmd-t10721.d-357B6D1E05DAA5A9B2827122C35E3FB1
│   ├── objs
│   └── rdmd.lock
```

Reply:
I'll merge and patch it, it's poorly named.